### PR TITLE
Bump DeDup for stats fix and DamageProfiler to latest version

### LINKED ISSUE #560: DeDup stats reporting is incorrect when unmapped reads included (the default in EAGER)
See https://github.com/apeltzer/DeDup/issues/10

### PR DESCRIPTION
**Pull request by jfy133**
# nf-core/eager pull request

This bumps DeDup to latest bug fix version to close #560 .

This also adds the number of covered by each individual and total SNPs in the eigenstrat genotype dataset produced by pileupcaller, when that is used for genotyping, in the final MutiQC report. 

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If necessary, also make a PR on the [nf-core/eager branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/eager)
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --paired_end`).
- [x] Make sure your code lints ([`nf-core lint .`](https://nf-co.re/tools)).
- [x] Documentation in `docs` is updated
- [x] `CHANGELOG.md` is updated
- [x] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/eager/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by jfy133** [comment]
Waiting for @TCLamnidis 's additions

**Comment by TCLamnidis** [comment]
required version of eigenstratdatabasetools is not yet updated in bioconda. should be up soon. Will then update README and CHANGELOG to force checks to restart

File: main.nf [hunk 1]
```diff
@@ -2614,7 +2614,32 @@ if (params.pileupcaller_snpfile.isEmpty ()) {
...
+   tuple samplename, libraryid, lane, seqtype, organism, strandedness, udg, path("*") from ch_for_eigenstrat_snp_coverage.dump()
+   
+   output:
+   tuple samplename, libraryid, lane, seqtype, organism, strandedness, udg, path("*.json") into ch_eigenstrat_snp_cov_for_multiqc
+   path("*_eigenstrat_coverage.txt")
+   
+   script:
+/*   """
+   eigenstrat_snp_coverage -i pileupcaller.${strandedness} -s ".txt" >${strandedness}_eigenstrat_coverage.txt -j ${strandedness}_eigenstrat_coverage_mqc.json
+   """*/
```
Comment (jfy133) [inline main.nf:2636]: Please remove if not necessary anymore

**Review by jfy133 — COMMENTED** [review]
One comment but otherwise looks good.

Update changelog/readme and feel free to approve and merge @TCLamnidis

File: main.nf (hunk 1 above)
Comment (TCLamnidis) [reply main.nf:2636]: That is what the command will become after the MQC release

Comment (TCLamnidis) [reply main.nf:2636]: at which point `parse_snp_coverage.py` will become redundant

Comment (jfy133) [reply main.nf:2636]: Ok please leave a description with that, so we don't forget!

**Review by TCLamnidis — APPROVED** [review]

**Pull request merged**